### PR TITLE
Sieve of Eratosthenes added

### PR DESCRIPTION
**Pull request by lavishsaluja**
Fixes https://github.com/sahilbansal17/Competitive_Coding/issues/45

### CODE REVIEW

File: Algorithms/NumberTheory/SieveOfEratosthenes.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Sieve Of Eratosthenes
+// Time Complexity O(nlog(log(n))) whereas following brute force will end you up with O(n*sqrt(n))
+#include <iostream>
+#include<math.h>
+using namespace std;
+
+int main(void){
+    int n;cin>>n;
```
Comment (arnav-mandal1234) [inline Algorithms/NumberTheory/SieveOfEratosthenes.cpp]: put it in next line.
edit all statements like n=4 to n = 4;
for loop write like this for (int i = 0; i < n; i ++)

File: Algorithms/NumberTheory/SieveOfEratosthenes.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    prime[1]=0;//since 0 & 1 are not prime ofcourse
+    /*
+    Here we will be running are first loop only till sqrt(n) instead of n because we know that
+    if a number is prime it does not have any factor till its square root so we
+    can use it for optimization.
+    */
+    for(int i=2;i<=sqrt(n);i++){
+        if(prime[i]==1){//if i is prime
+            for(int j=2;j*i<=n;j++){
+                prime[i*j]=0;//setting the multiples of i to 0 because they can't be prime ofcourse
```
Comment (arnav-mandal1234) [inline Algorithms/NumberTheory/SieveOfEratosthenes.cpp]: Comments should be on top of a line. not sideways

File: Algorithms/NumberTheory/SieveOfEratosthenes.cpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            }
+        }
+    }
+    for(int i=0;i<=n;i++){
+        if(prime[i]!=0){
+            cout<<i<<endl;
+        }
+    }
+	return 0;
```
Comment (arnav-mandal1234) [inline Algorithms/NumberTheory/SieveOfEratosthenes.cpp]: indentation fault

**Comment by lavishsaluja** [comment]
@arnav-mandal1234

File: Algorithms/NumberTheory/SieveOfEratosthenes.cpp (hunk 1 above)
Comment (arnav-mandal1234) [reply Algorithms/NumberTheory/SieveOfEratosthenes.cpp]: @lavishsaluja  edit your code with these changes.

**Pull request merged**